x-pack/legacy/plugins/reporting/export_types/csv/server/lib/hit_iterator.ts

```diff
@@ -3,18 +3,33 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import { SearchParams, SearchResponse } from 'elasticsearch';
 
+import { get } from 'lodash';
+import { SearchParams, SearchResponse } from 'elasticsearch';
 import { i18n } from '@kbn/i18n';
 import { CancellationToken, ScrollConfig, Logger } from '../../../../types';
 
-async function parseResponse(request: SearchResponse<any>) {
+interface Hit {
+  _source: any;
+  fields?: any;
+  highlight?: any;
+  inner_hits?: any;
+  matched_queries?: string[] | undefined;
+  sort?: string[] | undefined;
+}
+
+interface ParsedResponse {
+  scrollId: string | undefined;
+  hits: Hit[] | undefined;
+}
+
+async function parseResponse(request: Promise<SearchResponse<any>>): Promise<ParsedResponse> {
   const response = await request;
-  if (!response || !response._scroll_id) {
```

Comment:
I found documentation on the Javascript ES client that talks about creating a generator function that uses scan/scroll. The change made here aligns the functionality to that documentation, which also treats missing `._scroll_id` as a non-error:

https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/6.x/scroll_examples.html

Reply:
I believe this is the new client which Kibana does not yet support. We're still using 16.x of the legacy client.